### PR TITLE
fix: unintended error message from `get_logger_placeholder()`

### PR DESCRIPTION
**Pull request by cbscsm**
### Background
- Currently, `Litestar.get_logger` is `get_logger_placeholder` only when the app initialized by keyword argument `logging_config=None`.
  - We using this keyword argument as workaround for an issue like #2862
- `request.logger` is a property that returns `self.app.get_logger()`. It calls `Litestar.get_logger` with no argument. In standard library or picologging, it is allowed and will return root logger.
- code: https://github.com/litestar-org/litestar/blob/d8322b7a588a0a272d8b611d7d0c6ab1d2ccc326/litestar/connection/base.py#L268-L278
- But `get_logger_placeholder` doesn't allow calling it with no argument. So, when you initialize app with no logging_config and try to use `request.logger`, the error is not as intended.
- Actual
```text
TypeError: get_logger_placeholder() missing 1 required positional argument: '_'
```
- Expected
```text
ImproperlyConfiguredException: cannot call '.get_logger' without passing 'logging_config' to the Litestar constructor first
```

### Changes suggested
- So I suggest to add default value `None`.

### CODE REVIEW

**Comment by guacs** [comment]
@cbscsm thanks for this! I think this makes sense since it's the same for the standard library `getLogger` function as well. Could you add a test for this as well?

**Comment by cbscsm** [comment]
I added a test case to check if right exception raised when get_logger_placeholder called without args.

File: tests/unit/test_logging/test_logging_config.py [hunk 1]
```diff
@@ -82,6 +83,12 @@ def test_picologging_dictconfig_when_disabled(dict_config_mock: Mock) -> None:
         assert not dict_config_mock.called
 
 
+def test_get_logger_without_logging_config() -> None:
+    with create_test_client(logging_config=None) as client:
+        with pytest.raises(ImproperlyConfiguredException):
```
Comment (guacs) [inline tests/unit/test_logging/test_logging_config.py:88]: Could you add a `match` to make sure the exception message is the logging one? Just to confirm that the error is from the placeholder function.

**Review by guacs — CHANGES_REQUESTED** [review]

**Review by guacs — APPROVED** [review]
Thank you!

**Comment by guacs** [comment]
@all-contributors add @cbscsm for code, test

**Pull request merged**